datafusion/core/tests/user_defined/user_defined_scalar_functions.rs

```diff
@@ -483,6 +485,188 @@ async fn test_user_defined_functions_with_alias() -> Result<()> {
     Ok(())
 }
 
+struct AddIndexToStringScalarUDF {
```

Comment:
Perhaps we can add some comments here explaining what this is:

```suggestion
/// Volatile UDF that should be append a different value to each row
struct AddIndexToStringScalarUDF {
```